android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -734,18 +609,11 @@ public static boolean containsLexicalModel(Context context, String lexicalModelK
   protected static int getCurrentKeyboardIndex(Context context) {
     int index = -1;
 
-    if (keyboardsList == null) {
-      keyboardsList = getKeyboardsList(context);
-    }
-
+    List<Keyboard> keyboardsList = KeyboardController.getInstance().get();
     if (keyboardsList != null) {
-      int length = keyboardsList.size();
-      for (int i = 0; i < length; i++) {
-        HashMap<String, String> kbInfo = keyboardsList.get(i);
-        String langId = kbInfo.get(KMManager.KMKey_LanguageID);
-        String kbId = kbInfo.get(KMManager.KMKey_KeyboardID);
-        String kbKey = String.format("%s_%s", langId, kbId);
-        if (kbKey.equals(KMKeyboard.currentKeyboard())) {
+      for(int i=0; i<keyboardsList.size(); i++) {
```

Comment:
It'd be nicer to have a `KeyboardList.getIndexOfKey()` function

Reply:
I've added `KeyboardController.getKeyboardIndex()`